client/tests/integration/account_registration/private.rs

```diff
@@ -0,0 +1,90 @@
+//! In private use cases, a new account should be:
+//! - recognized when targeted as (a part of) a destination (or an object) of a creative instruction
+//!     - becomes able to hold assets, permissions, roles, and metadata
+//!     - cannot yet execute any queries or transactions
+//! - activated when manually registered by an administrative account
+//!     - becomes able to execute queries and transactions
+
+use iroha_client::data_model::prelude::*;
+use test_network::*;
+use test_samples::{gen_account_in, ALICE_ID};
+
+/// A new account e.g. "carol" should be:
+/// - recognized when targeted as a destination of a transfer of numeric asset e.g. "rose"
+/// - activated when manually registered by an administrative account e.g. "alice"
+///
+/// # Scenario
+///
+/// 0. new carol targeted ... carol recognized
+/// 0. carol tries query ... err
+/// 0. carol tries transaction ... err
+/// 0. register carol ... carol activated
+/// 0. carol tries query ... ok
+/// 0. carol tries transaction ... ok
```

Comment:
For private scenarios, the aim is no impact on membership and privacy while relaxing restrictions on the order of registering and targeting accounts